src/java.base/share/classes/java/lang/StringLatin1.java

```diff
@@ -760,8 +760,7 @@ public static String newString(byte[] val, int index, int len) {
         if (len == 0) {
             return "";
         }
-        return new String(Arrays.copyOfRange(val, index, index + len),
-                          LATIN1);
+        return new String(String.copyBytes(val, index, len), LATIN1);
```

Comment:
It might be worthwhile putting a comment at the top of this method saying that the caller is required to bounds-check the arguments. It's called from several other classes in this package, so it would be good to document this package-internal contract. I checked the callers and they seem fine.